SteamKit2/SteamKit2/Steam/CDN/Client.cs

```diff
@@ -81,11 +83,60 @@ public async Task<DepotManifest> DownloadManifestAsync( uint depotId, ulong mani
                 url = $"depot/{depotId}/manifest/{manifestId}/{MANIFEST_VERSION}";
             }
 
-            var manifestData = await DoRawCommandAsync( server, url, proxyServer ).ConfigureAwait( false );
+            using var request = new HttpRequestMessage( HttpMethod.Get, BuildCommand( server, url, proxyServer ) );
 
-            manifestData = ZipUtil.Decompress( manifestData );
+            using var cts = new CancellationTokenSource();
+            cts.CancelAfter( RequestTimeout );
+
+            DepotManifest depotManifest;
+
+            try
+            {
+                using var response = await httpClient.SendAsync( request, HttpCompletionOption.ResponseHeadersRead, cts.Token ).ConfigureAwait( false );
```

Comment:
Because the cts and response are local, I can't just return the response.Content. The only other code path using DoRawCommandAsync is the chunk download, and that code should also stream.

The chunk download has other stupid things like the vzip decompress using LZMA which allocates a 1mb buffer whenever the decoder properties are set.

And the rented buffer would have to be copied if chunk is downloaded without giving a depot key.